app/presenters/comment_presenter.rb

```diff
@@ -0,0 +1,46 @@
+class CommentPresenter < ApplicationPresenter
+  presents :comment
+  delegate :text, to: :comment
+
+  def submitters_name
+    comment.submitter.name
+  end
+
+  def submitters_link
+    if h.user_signed_in?
+      h.comment_reply_comment_path(comment)
+    else
+      h.user_path(comment.submitter.name)
+    end
+  end
+
+  def link_prefix
+    if h.user_signed_in?
+      "Reply to "
+    else
+      " - contributed by "
+    end
+  end
+
+  def text_trailer
+    h.link_to (link_prefix + submitters_name), submitters_link
+  end
+
+  def score_present?
+    comment&.votes.present?
+  end
+
+  def sum
+    if score_present?
+      comment&.votes.sum(:value)
+    end
+  end
```

Comment:
does it make sense for this method to return `0` instead of `nil`?

Reply:
I guess having a `0` rendered cuts down the bounce effect that can happen the first time a thing is voted on